subprojects/soak/src/integTest/groovy/org/gradle/jvm/toolchain/JavaToolchainDownloadSoakTest.groovy

```diff
@@ -79,6 +82,41 @@ class JavaToolchainDownloadSoakTest extends AbstractIntegrationSpec {
         assertJdkWasDownloaded("openj9")
     }
 
+    def "clean destination folder when downloading toolchain"() {
+        when:
+        result = executer
+                .withTasks("compileJava", "-Porg.gradle.java.installations.auto-detect=false")
+                .expectDocumentedDeprecationWarning("Java toolchain auto-provisioning needed, but no java toolchain repositories declared by the build. Will rely on the built-in repository. " +
+                        "This behavior has been deprecated. This behavior is scheduled to be removed in Gradle 8.0. " +
+                        "In order to declare a repository for java toolchains, you must edit your settings script and add one via the toolchainManagement block. " +
+                        "See https://docs.gradle.org/current/userguide/toolchains.html#sec:provisioning for more details.")
+                .run()
+
+        then:
+        javaClassFile("Foo.class").assertExists()
+        assertJdkWasDownloaded("adoptopenjdk")
+
+        when:
+        //delete marker file to make the previously downloaded installation undetectable
+        def markerFile = findMarkerFile(executer.gradleUserHomeDir.file("jdks"))
+        markerFile.delete()
+
+        then:
+        !markerFile.exists()
+
+        when:
+        executer
+                .withTasks("compileJava", "-Porg.gradle.java.installations.auto-detect=false", "-Porg.gradle.java.installations.auto-download=true")
+                .expectDocumentedDeprecationWarning("Java toolchain auto-provisioning needed, but no java toolchain repositories declared by the build. Will rely on the built-in repository. " +
+                        "This behavior has been deprecated. This behavior is scheduled to be removed in Gradle 8.0. " +
+                        "In order to declare a repository for java toolchains, you must edit your settings script and add one via the toolchainManagement block. " +
+                        "See https://docs.gradle.org/current/userguide/toolchains.html#sec:provisioning for more details.")
+                .run()
+
+        then:
+        markerFile.exists()
```

Comment:
It's not clear to me how this test is assuring that the directory is actually cleaned.  What is the behavior now if the toolchain directory already exists?  Does it error?  In other words, if somebody changes things tomorrow such that it no longer cleans out the directory, would this catch it?

Reply:
Without the directory cleaning the test produces the same error as the one that prompted me to do this change, which is a write permission error, like this:

```
Caused by: org.gradle.api.internal.file.AbstractFileTreeElement$CopyFileElementException: Could not copy file '/Users/jozsef/Work/gradle/subprojects/soak/build/tmp/test files/JavaToolcha.Test/4d0ov/user-home/jdks/OpenJDK14U-jdk_x64_mac_hotspot_14/jdk-14.0.2+12/Contents/Home/lib/server/classes.jsa' to '/Users/jozsef/Work/gradle/subprojects/soak/build/tmp/test files/JavaToolcha.Test/4d0ov/user-home/jdks/adoptopenjdk-14-x86_64-os_x/jdk-14.0.2+12/Contents/Home/lib/server/classes.jsa'.
	at org.gradle.api.internal.file.AbstractFileTreeElement.copyTo(AbstractFileTreeElement.java:88)
	at org.gradle.api.internal.file.copy.DefaultFileCopyDetails.copyTo(DefaultFileCopyDetails.java:131)
	at org.gradle.api.internal.file.copy.FileCopyAction$FileCopyDetailsInternalAction.processFile(FileCopyAction.java:49)
	at org.gradle.api.internal.file.copy.NormalizingCopyActionDecorator.lambda$execute$0(NormalizingCopyActionDecorator.java:70)
	at org.gradle.api.internal.file.copy.DuplicateHandlingCopyActionDecorator.lambda$execute$0(DuplicateHandlingCopyActionDecorator.java:65)
	at org.gradle.api.internal.file.copy.CopyFileVisitorImpl.processFile(CopyFileVisitorImpl.java:68)
   ... 165 more
Caused by: java.io.FileNotFoundException: /Users/jozsef/Work/gradle/subprojects/soak/build/tmp/test files/JavaToolcha.Test/4d0ov/user-home/jdks/adoptopenjdk-14-x86_64-os_x/jdk-14.0.2+12/Contents/Home/lib/server/classes.jsa (Permission denied)
	at org.gradle.api.internal.file.AbstractFileTreeElement.copyFile(AbstractFileTreeElement.java:100)
	at org.gradle.api.internal.file.AbstractFileTreeElement.copyTo(AbstractFileTreeElement.java:83)
	... 207 more
```